lib/ZumoOta/src/BootloaderCom.cpp

```diff
@@ -95,28 +103,246 @@ class IntProgCmdProvider: public CmdProvider {
+    
+    CommandInfo m_updatedCmd; /**<Intermediate storage for updated commands.*/
+
+    const char* m_fileName = "/firmware.bin"; /**< Name of the file to be read.*/
+    uint8_t m_firmwareBuffer[128]; /**< Buffer to hold firmware data read from the file.*/
+    size_t m_firmwareBytesRead; /**< Number of bytes read from the firmware file.*/
+    File m_firmwareFile; /**< Handle for the firmware file.*/
+
+    /**
+     * @brief Enumeration of programming commands.
+     * This enumeration defines the various programming commands used during firmware programming.
+     * It includes commands for setting memory addresses and writing data blocks.
+     */
+   enum ProgCmds
+   {
+        FLSPRG_SET_ADDR,    /**< Command to set the memory address. */
+        FLSPRG_WRITE_BLOCK,  /**< Command to write a data block to memory.*/
+        FLSPRG_COMPLETE   /**<Signalize the end of the Flashing.*/
+    };
+
+    /** 
+    * @brief The next programming command to be executed. 
+    * This variable represents the next programming command to be executed in the sequence.
+    * It initializes with FLSPRG_SET_ADDR, typically used to set the memory address first.
+    */
+    ProgCmds m_nextCmd = FLSPRG_SET_ADDR;
+    uint8_t m_updatedCmdBuffer[WRITE_BLOCK_CMD_LENGTH + FLASH_BLOCK_LENGTH]; /**<Buffer for updated commands.*/
+
+    /**
+     * @brief Current memory address for writing data during firmware programming. 
+     * This variable represents the current memory address where data will be written
+     * during the firmware programming process. It is used to keep track of the
+     * memory address for sequential data writes.
+     */
+    uint16_t  m_currWriteMemAddr; 
+
+   /**
+    * @brief Updates the command.
+    * This method updates the provided command based on specific requirements
+    * before it is sent for execution. It modifies the command buffer and size
+    * to reflect the changes required by the bootloader communication protocol.
+    * @param fwProgCmd Pointer to the CommandInfo structure representing the command to be updated.
+    * The method modifies the command buffer and size to reflect the changes.
+    * @remarks This method is called before sending each command to the bootloader for execution.
+    * The update process includes modifying command identifiers and parameters to adhere to the
+    * bootloader communication protocol requirements.
+    */
+    bool updateFwProgCmd(CommandInfo& fwProgCmd)
+    {
+        if (m_nextCmd == FLSPRG_SET_ADDR)
+        {
+            /*Perform the required updates.*/
+            m_updatedCmdBuffer[0] = 0x41;  /**< The first value is always 0x41.*/
+            m_updatedCmdBuffer[1] = (m_currWriteMemAddr >> 8) & 0xFF; /**<Higher 8 bits of m_currWriteMemAddr.*/
+            m_updatedCmdBuffer[2] = m_currWriteMemAddr & 0xFF; //**<Lower 8 bits of m_currWriteMemAddr.*/
+            /*Increase m_currWriteMemAddr by 64 for the next command.*/
+            m_currWriteMemAddr += 64U;

```

Comment:
64 - really, why ?

Reply:
Each address addresses one WORD with 2(!) bytes, this is why one page consists of 64 addresses but 128 bytes memory

Reply:
Strange. Haven't seen such logic bevor.

Then you should use "FLASH_BLOCK_LENGTH / 2U" instead of 64. Can be next commit

Reply:
Okay!